cmd/sgbouncewizard/main.go

```diff
@@ -1,39 +1,120 @@
 package main
 
 import (
+	"context"
+	"database/sql"
 	"encoding/json"
+	"fmt"
 	"log"
 	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	_ "github.com/go-sql-driver/mysql"
 
-	"github.com/jimmyjames85/bouncecm/internal/db"
-	"github.com/jimmyjames85/bouncecm/internal/models"
 	"github.com/go-chi/chi"
 	"github.com/go-chi/chi/middleware"
+	"github.com/jimmyjames85/bouncecm/internal/db"
+	"github.com/jimmyjames85/bouncecm/internal/models"
 	"golang.org/x/crypto/bcrypt"
 )
 
+type BounceRule struct {
+	Id            int     `json:"id"`
+	Response_code string  `json:"response_code"`
+	Enhanced_code string  `json:"enhanced_code"`
+	Regex         string  `json:"regex"`
+	Priority      int     `json:"priority"`
+	Description   *string `json:"description"`
+	Bounce_action string  `json:"bounce_action"`
+}
+
+type RulesObject struct {
+	Rules    []BounceRule `json:"rules"`
+	NumRules int          `json:"numRules"`
+}
+
 func main() {
+
 	r := chi.NewRouter()
 
 	r.Use(middleware.RequestID)
 	r.Use(middleware.RealIP)
 	r.Use(middleware.Logger)
 	r.Use(middleware.Recoverer)
 
-	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("welcome"))
+	r.Route("/bounce_rules", func(r chi.Router) {
+		r.Get("/", ListRules)
+		r.Post("/", createRule)
+
+		r.Route("/{bounce_id}", func(r chi.Router) {
+			r.Use(RuleContext)
+			r.Get("/", getRule)
+			r.Delete("/", deleteRule)
+			r.Put("/", updateRule)
+		})
 	})
 
 	r.Route("/user", func(r chi.Router) {
 		r.Post("/", CheckUser)
 	})
 
-	r.Route("/bounce_rules", func(r chi.Router) {
-		r.Get("/", ListRules)
+	http.ListenAndServe(":3000", r)
+}
+
+func RuleContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var rule *BounceRule
+		var err error
+		if bounce_id := chi.URLParam(r, "bounce_id"); bounce_id != "" {
+			rule, err = getRuleDB(bounce_id)
+			checkErr(err)
+		}
+		ctx := context.WithValue(r.Context(), "rule", rule)
+		next.ServeHTTP(w, r.WithContext(ctx))
 	})
+}
 
-	http.ListenAndServe(":3000", r)
+func getRule(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	rule := r.Context().Value("rule").(*BounceRule)
+	data, err := json.Marshal(rule)
+	checkErr(err)
```

Comment:
If there is an err, we are not guaranteed `data` is populated. We should handle this as `http.StatusInternalServerError`.